internal/controller/controllers/crd_utils.go

```diff
@@ -40,6 +43,15 @@ func (u *CRDUtils) CreateAgentCR(ctx context.Context, log logrus.FieldLogger, ho
 	}
 
 	if k8serrors.IsNotFound(err) {
+		infraEnv, err1 := u.getInfraEnvByCluster(ctx, log, clusterNamespace, clusterName)
+		if err1 != nil {
+			return errors.Wrap(err, fmt.Sprintf("Failed to search an InfraEnv for Cluster %s", clusterName))
```

Comment:
you can use wrapf then you will not need sprintf

Reply:
done